rocAL/rocAL/source/video_read_and_decode.cpp

```diff
@@ -0,0 +1,303 @@
+}
+
+void VideoReadAndDecode::reset()
+{
+    _reader->reset();
+}
+
+size_t
+VideoReadAndDecode::count()
+{
+    return _reader->count();
+}
+
+float VideoReadAndDecode::convert_framenum_to_timestamp(size_t frame_number)
+{
+    float timestamp;
+    timestamp = (float)frame_number / _frame_rate;
+    return timestamp;
+}
+
+void VideoReadAndDecode::decode_sequence(size_t sequence_index)
+{
+    if (_video_decoder[_sequence_video_idx[sequence_index]]->Decode(_decompressed_buff_ptrs[sequence_index], _start_frame[sequence_index], _sequence_length, _stride, _max_decoded_width, _max_decoded_height, 
+        _max_decoded_stride, _out_pix_fmt) == VideoDecoder::Status::OK)
+    {
+        for (size_t s = 0; s < _sequence_length; s++)
+        {
+            _actual_decoded_width[(sequence_index * _sequence_length) + s] = _max_decoded_width;
+            _actual_decoded_height[(sequence_index * _sequence_length) + s] = _max_decoded_height;
+        }
+    }
+}
+
+VideoLoaderModuleStatus
+VideoReadAndDecode::load(unsigned char *buff,
+                         std::vector<std::string> &names,
+                         const size_t max_decoded_width,
+                         const size_t max_decoded_height,
+                         std::vector<uint32_t> &roi_width,
+                         std::vector<uint32_t> &roi_height,
+                         std::vector<uint32_t> &actual_width,
+                         std::vector<uint32_t> &actual_height,
+                         std::vector<std::vector<size_t>> &sequence_start_framenum_vec,
+                         std::vector<std::vector<std::vector<float>>> &sequence_frame_timestamps_vec,
+                         RaliColorFormat output_color_format)
+{
+    if (max_decoded_width == 0 || max_decoded_height == 0)
+        THROW("Zero image dimension is not valid")
+    if (!buff)
+        THROW("Null pointer passed as output buffer")
+    if (_reader->count() < _batch_size)
+        return VideoLoaderModuleStatus::NO_MORE_DATA_TO_READ;
+
+    std::vector<size_t> sequence_start_framenum;
+    std::vector<std::vector<float>> sequence_frame_timestamps;
+    sequence_start_framenum.resize(_sequence_count);
+    sequence_frame_timestamps.resize(_sequence_count);
+    for (size_t it = 0; it < (_sequence_count); it++)
+        sequence_frame_timestamps[it].resize(_sequence_length);
+
+    const auto ret = video_interpret_color_format(output_color_format);
+    const unsigned output_planes = std::get<1>(ret);
+    _out_pix_fmt = std::get<2>(ret);
+    const size_t image_size = max_decoded_width * max_decoded_height * output_planes * sizeof(unsigned char);
+    _max_decoded_width = max_decoded_width;
+    _max_decoded_height = max_decoded_height;
+    _max_decoded_stride = max_decoded_width * output_planes;
+
+    // File read is done serially since I/O parallelization does not work very well.
+    _file_load_time.start(); // Debug timing
+
+    size_t fsize = 1280 * 720 * 3;
```

Comment:
why is this hardcoded?

Reply:
This will be removed when we introduce a separate VideoReader